src/mz/src/utils.rs

```diff
@@ -88,3 +88,39 @@ pub(crate) fn exit_with_fail_message(message: ExitMessage) -> ! {
     }
     exit(1);
 }
+
+/// Layouts an UUID.
+/// Frontegg API needs the UUID with the layout.
+fn layout_uuid(uuid: String) -> String {
+    let low_time = uuid[0..8].to_string();
+    let mid_time = uuid[8..12].to_string();
+    let high_time = uuid[12..16].to_string();
+    let clock_sequence = uuid[16..20].to_string();
+    let node = uuid[20..32].to_string();
+
+    format!(
+        "{}-{}-{}-{}-{}",
+        low_time, mid_time, high_time, clock_sequence, node
+    )
+}
+
+/// Turns the app-password into an API token
+pub(crate) fn api_token_from_password(password: String) -> Option<FronteggAPIToken> {
+    if password.len() != 68 || !password.starts_with("mzp_") {
+        None
+    } else {
+        let client_id = layout_uuid(password[4..36].to_string());
+        let secret = layout_uuid(password[36..68].to_string());
+        Some(FronteggAPIToken { client_id, secret })
+    }
+}
+
+/// Format a Materialize password using the client_id and secret
+pub(crate) fn format_password(client_id: String, secret: String) -> String {
```

Comment:
Make this a method on `FronteggAPIToken` instead? That way, you don't have to get the strings to correctly line up.